rumqttd/src/router/routing.rs

```diff
@@ -761,8 +785,22 @@ impl Router {
                             break;
                         }
                     };
+                    self.scheduler.reschedule(id, ScheduleReason::IncomingAck);
+                }
+                Packet::PubComp(pubcomp, _) => {
+                    info!(pkid = pubcomp.pkid, "received pubcomp");
+                    let outgoing = self.obufs.get_mut(id).unwrap();
+                    let pkid = pubcomp.pkid;
+                    if outgoing.register_pubcomp(pkid) {
+                        error!(
+                            pkid,
+                            "ack received for pkid {}, but the pkid didn't exists!", pkid
+                        );
+                        // shall we ignore it or diconnect the client?
+                        // disconnect = true;
+                        // break;
```

Comment:
yeah to be consistent with other places, we should disconnect the client as there might be some state mismatch if we just ignore it.

Reply:
okie!